src/WpfMath/TexFormulaParser.cs

```diff
@@ -434,6 +439,14 @@ private static SourceSpan ReadElement(SourceSpan value, ref int position)
                     }
             }
 
+            if (_commandRegistry.TryGetValue(command, out var parser))
+            {
+                var context = new CommandContext(this, formula, value, start, position);
+                var parseResult = parser.ProcessCommand(context);
+                position = parseResult.NextPosition;
```

Comment:
Let's add an assertion here: `parseResult.NextPosition > position`; otherwise the parser didn't made any progress.

Also, add a test that declares a defunct parser and checks if an assertion works.

Reply:
Double it. Another solution is introduce assertion to command parser.

Reply:
I want the command parser to stay an interface. In future, it will be an external-visible API (so the users may create their own parsers), so I'd like to currently concentrate the checks on "our" side of the API. So we can't make the check in the command parser (especially in its current state when there's no common code between parsers, because we only have one parser).

Reply:
OK, let’s do that.